### PR TITLE
Rename hash to mapping and array to sequence

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

**Comment by xabbuh** [comment]
There's also a variable named `hash` in `tests/Fixtures/regression/strings_like_numbers.test`.

File: doc/filters/merge.rst [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ``merge``
 =========
 
-The ``merge`` filter merges an array with another array:
+The ``merge`` filter merges sequences and mappings.
+
+The ``merge`` filter also works on sequences:
```
Comment (xabbuh) [inline doc/filters/merge.rst]: The "also" here and below reads weird. What about merging both code blocks instead?

Comment (fabpot) [reply doc/filters/merge.rst]: I've done it a bit differently, it's better now.

**Comment by fabpot** [comment]
Should be good now

**Pull request merged**